### ISSUE TITLE
Conflicting regions in provider block causes mixed region application/destruction issues

### ISSUE THREAD

**Issue by mschuchard** [hashicorp/terraform-provider-aws#7442]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Terraform: 0.11.11
AWS: 1.57

### Affected Resource(s)

provider aws

### Terraform Configuration Files

```hcl
provider "aws" {
  version = "~> 1.57"
  region  = "us-west-2"
  profile = "personal" #<-- contains config for us-east-1
}
```

### Debug Output

N/A

### Panic Output

N/A

### Expected Behavior

Resources would only be applied in one region (one overrides the other), and/or there would be an alert on attempting to apply with two conflicting regions specified in the provider.

### Actual Behavior

Terraform applies some resources in the specified profile region `us-east-1` AND some resources in the specified region `us-west-2`.

### Steps to Reproduce

1. `terraform init`
2. `terraform apply` <-- applied some resources in us-east-1 and some in us-west-2
3. `terraform destroy` <-- for resources in `us-east-1`, they were destroyed or there was a timeout on destroying them; for resources in `us-west-2`, Terraform claimed they were destroyed when they were actually not or there was a timeout on destroying them

### Important Factoids

N/A

### References

Originally mistakenly filed as hashicorp/terraform#20210

* #0000

**Comment by ringods** [comment]
@mschuchard can you test again with `region=us-east-1` removed from your `~/.aws/config`?

I just filed #7750 with the assumption that `region` is *never* read from your profile and you are hitting the default value:

https://github.com/terraform-providers/terraform-provider-aws/blob/687f09ebf7d2b5d0fe1b92f45c3c49ead8cb3711/aws/provider.go#L60-L69

**Comment by mschuchard** [comment]
If I removed the region from the config, then there would be no conflicting regions, no bug, and a completely different situation than this issue describes.

**Comment by kerryhatcher** [comment]
Using the versions below I'm getting a similar issue. Consecutive runs of plan with no changes to anything whatsoever will sometimes result in an error that my west VPC doesn't exist. Checking AWS Cloudtrail I noticed that sometimes (and totally randomly) Terraform is trying to query or apply west resources to the east region. I've attached a screenshot of the logs to show how random it seems. 

The strange part is I use a VPC data source in the module. So I pass in a string with the VPC id to the module, VPC data source will find the VPC and pass that along to the SG resource, then the SG resource will fail. Other times the VPC data source will fail before it ever tries to make the SG. Like I said, I can run the plan and apply commands over and over and each time will be a random result with no changes to TF code, AWS, ENV vars, etc. 

It does appear to somehow be related to my .aws settings file and/or the AWS_DEFAULT_REGION/AWS_REGION env vars. The error appears to go away when I emptied `~/.aws/config`, started a new terminal (with none of the region env vars), and used a clean workspace. Switching back to a previously used workspace seemed to show the bug again even after removing the config & env vars. So somehow there is something getting cached in the state file and causing issues. 


```
Terraform v0.12.28
+ provider.archive v1.3.0
+ provider.aws v2.68.0
+ provider.null v2.1.2
```

![image](https://user-images.githubusercontent.com/1011814/86199090-31180d80-bb27-11ea-9b93-920950f62d81.png)
